### PR TITLE
cleanup: remove deprecated targets and rules

### PR DESCRIPTION
**Pull request by coryan**
And the CI scripts used to test them.  Part of the work for #5726

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8413)

### CODE REVIEW

**Review by dbolduc — CHANGES_REQUESTED** [review]
I think we need a CHANGLEOG.md update to move [this](https://github.com/googleapis/google-cloud-cpp/blame/main/CHANGELOG.md#L7-L40) out of future breaking change and into the current release's breaking changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -87,13 +87,13 @@ https://github.com/googleapis/google-cloud-cpp/issues/8234.
         `pubsub_client`, `storage_client`, `spanner_client`.
       - Some target aliases, including `bigtable::client`, `bigtable::protos`
   - **pkg-config users:** applications should use the modules starting with
-    `google_cloud_cpp`. All other modules are not retired
+    `google_cloud_cpp`. All other modules are not retired.
```
Comment (dbolduc) [inline CHANGELOG.md:90]: super nit: maybe avoid the double negative here. "No other modules are retired."

File: CHANGELOG.md [hunk 2]
```diff
@@ -87,13 +87,13 @@ https://github.com/googleapis/google-cloud-cpp/issues/8234.
...
   - **Direct users of -l${library} flags:** we do not recommend that
     applications uses `-l` flags directly, please use `pkg-config` and/or
     the target names under CMake or Bazel. We make this recommendation because
     we do not know of any mechanism to provide backwards compatibility for such
     flags.
-  - More details about the rationale for these change in [#5726]
+  - More details about the rationale for these change in [#5726].
```
Comment (dbolduc) [inline CHANGELOG.md:96]: s/these change/these changes/

**Review by dbolduc — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (dbolduc) [reply CHANGELOG.md]: actually, it's not a double negative. I still prefer saying `¬∃ m ∈ other_modules | m.is_retired` to `∀ m ∈ other_modules | ¬m.is_retired` though

Comment (coryan) [reply CHANGELOG.md]: Ack, but all I was trying to say is "now retired"

**Pull request merged**